tests/unit/model/designation-test.js

```diff
@@ -54,3 +54,191 @@ test('Relationships with other models', function(assert){
+  var store = this.store();
+  var item1 = null;
+  var item2 = null;
+  var item3 = null;
+  var despatchedItemsIds = null;
+
+  Ember.run(function(){
+    store.createRecord('item', { id: 1, sentOn: "12/07/2016" });
+    store.createRecord('item', { id: 2, sentOn: "12/07/2016" });
+    store.createRecord('item', { id: 3, sentOn: null });
+    item1 = store.peekRecord('item', 1);
+    item2 = store.peekRecord('item', 2);
+    item3 = store.peekRecord('item', 3);
+    model.get('items').pushObject(item1);
+    model.get('items').pushObject(item2);
+    model.get('items').pushObject(item3);
+  });
+  despatchedItemsIds = model.get('dispatchedItems').getEach('id');
+  assert.equal(despatchedItemsIds.get('length'), 2);
+  assert.equal(despatchedItemsIds[0], item1.get('id'));
+  assert.equal(despatchedItemsIds[1], item2.get('id'));
+});
+
+test('check designatedItems returns items with sentOn null', function(assert){
+  assert.expect(3);
+  const model = this.subject();
+  var store = this.store();
+  var item1 = null;
+  var item2 = null;
+  var item3 = null;
+  var designatedItemsIds = null;
+
+  Ember.run(function(){
+    store.createRecord('item', { id: 1, sentOn: null });
+    store.createRecord('item', { id: 2, sentOn: null });
+    store.createRecord('item', { id: 3, sentOn: "12/07/2016" });
+    item1 = store.peekRecord('item', 1);
+    item2 = store.peekRecord('item', 2);
+    item3 = store.peekRecord('item', 3);
+    model.get('items').pushObject(item1);
+    model.get('items').pushObject(item2);
+    model.get('items').pushObject(item3);
+  });
+  designatedItemsIds = model.get('designatedItems').getEach('id');
+  assert.equal(designatedItemsIds.get('length'), 2);
+  assert.equal(designatedItemsIds[0], item1.get('id'));
+  assert.equal(designatedItemsIds[1], item2.get('id'));
+});
+
+test('check designatedOrdersPackages returns only designated orders_packages', function(assert){
+  assert.expect(3);
+  const model = this.subject();
+  var store = this.store();
+  var orders_package1 = null;
+  var orders_package2 = null;
+  var orders_package3 = null;
+  var designatedOrdersPackagesIds = null;
+
+  Ember.run(function(){
+    store.createRecord('orders_package', {id: 1, state: 'designated'});
+    store.createRecord('orders_package', {id: 2, state: 'designated'});
+    store.createRecord('orders_package', {id: 3, state: 'dispatched'});
+
+    orders_package1 = store.peekRecord('orders_package',1);
+    orders_package2 = store.peekRecord('orders_package',2);
+    orders_package3 = store.peekRecord('orders_package',3);
+
+    model.get('ordersPackages').pushObject(orders_package1);
+    model.get('ordersPackages').pushObject(orders_package2);
+    model.get('ordersPackages').pushObject(orders_package3);
+  });
+
+  designatedOrdersPackagesIds = model.get('designatedOrdersPackages').getEach('id');
+
+  assert.equal(designatedOrdersPackagesIds.get('length'),2);
+  assert.equal(designatedOrdersPackagesIds[0], orders_package1.get('id'));
```

Comment:
@viveksriram1 Instead of comparing array indexes can we compare array with array?
Means assert.equal(despatchedItemsIds, itemIds); where itemIds will be an array of item ids.